src/waveform/renderers/allshader/waveformrendererpreroll.cpp

```diff
@@ -95,11 +117,15 @@ void WaveformRendererPreroll::paintGL() {
     // to indicate the respective zones.
     const bool preRollVisible = firstDisplayedPosition < 0;
     const bool postRollVisible = lastDisplayedPosition > 1;
+    const int numVerticesPerRectangle = 6;
 
-    if (!(preRollVisible || postRollVisible)) {
-        return;
+    if (!preRollVisible && !postRollVisible) {
+        return false;
     }
 
+    const int reserved = (preRollVisible ? numVerticesPerRectangle : 0) +
+            (postRollVisible ? numVerticesPerRectangle : 0);
```

Comment:
does it make sense to move closer to the first use? and potentially give it a better name?

Reply:
done